### PR TITLE
Update to v0.54.1

### PR DESCRIPTION
**Pull request by pseudoyim**

### CODE REVIEW

**Comment by pseudoyim** [comment]
Only the `osx-arm64` build is failing right now. Asked the Numba team about this and it may require further debugging.

Confirmed with Product that we are ok with merging this for now to update to `v0.54.1` for all other platforms.

**Review by varlackc — COMMENTED** [review]
Could we use the `skip` option in the mean time for `osx-arm64`?

We could probably do something along the lines of 
`skip: true. #[py<37 or osx-arm64]`

File: recipe/meta.yaml [hunk 1]
```diff
@@ -69,14 +69,12 @@ test:
     - scipy
     - ipython
     - setuptools
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:69]: Add please `pip` and `pip check` in commands

File: recipe/meta.yaml [hunk 2]
```diff
@@ -17,29 +17,31 @@ build:
...
     - {{ compiler('cxx') }}
     # llvm is needed for the headers
-    - llvm-openmp >=10.0  # [osx]
-
+    - llvm-openmp              # [osx]
 
   host:
     - python
     - pip
     - setuptools
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:39]: - Missing wheel package in requirements

File: recipe/meta.yaml [hunk 3]
```diff
@@ -69,14 +69,12 @@ test:
...
     # Need these for AOT. Do not init msvc as it may not be present
     - {{ compiler('c') }}      # [not win]
     - {{ compiler('cxx') }}    # [not win]
-    # 10 tests fail on NumPy 1.20, 1 test fails on 1.19
-    - numpy <1.21  # [py<39]
-    - numpy <1.21  # [py>=39]
+    - python <3.10
 
   imports:
     - numba
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:80]: I would add more imports:
- numba.cloudpikle
- numba.cpython
- numba.experimental
- numba.mics
- numba.parfors
- numba.typed
- numba.types

But it's not mandatory

File: recipe/meta.yaml [hunk 4]
```diff
@@ -17,29 +17,31 @@ build:
...
     # llvm is needed for the headers
-    - llvm-openmp >=10.0  # [osx]
-
+    - llvm-openmp              # [osx]
 
   host:
     - python
     - pip
     - setuptools
     - llvmlite 0.37.*
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:41]: max llvmlite <0.38 https://github.com/numba/numba/blob/39aef3deb073a67b0bf121e099ecf5b33d799f40/setup.py#L28

Comment (pseudoyim) [reply recipe/meta.yaml:41]: `llvmlite` is a different package from `llvm-openmp`.

File: recipe/meta.yaml (hunk 1 above)
Comment (pseudoyim) [reply recipe/meta.yaml:69]: Done!

File: recipe/meta.yaml (hunk 2 above)
Comment (pseudoyim) [reply recipe/meta.yaml:39]: Added wheel

File: recipe/meta.yaml (hunk 4 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:41]: I mean:
`- llvmlite >=0.37.0,<0.38`
But your line is also says the same

**Review by skupr-anaconda — APPROVED** [review]

**Review by varlackc — APPROVED** [review]

**Pull request merged**